src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -80,11 +81,37 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         validateStreamId(cursors, streamId, subscriptionId);
 
+        LOG.debug("[COMMIT_CURSORS] stream IDs validation finished");
+
+        final Iterator<EventTypePartition> distinctPartitions = cursors.stream()
+                .map(c -> new EventTypePartition(c.getEventType(), c.getPartition()))
+                .distinct()
```

Comment:
I meant that Java does not have an option in distinct method to provide a comparator, it would be very handy to distinct by specific function.